src/openms/source/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.cpp

```diff
@@ -154,6 +154,76 @@ namespace OpenMS
     background = avg_noise_level * nr_points;
   }
 
+  void MRMTransitionGroupPicker::calculateBgEstimationExact_(const MSChromatogram& chromatogram,
+      double best_left, double best_right, double peak_height, double & background, double & avg_noise_level)
+  {
+    // determine (in the chromatogram) the intensity at the left / right border
+    double intensity_left = 0.0;
+    double rt_apex = 0.0;
+    double intensity_right = 0.0;
+    
+    // calculate the average noise level
+    for (MSChromatogram::const_iterator it = std::next(chromatogram.begin()); it != chromatogram.end(); ++it)
+    {
+      MSChromatogram::const_iterator it_prev = it;
+      --it_prev; //previous point
+
+      if (it->getMZ() >= best_left && it_prev->getMZ() < best_left)
+      {
+        intensity_left = it->getIntensity();
+      }
+      else if (it->getIntensity() >= peak_height && it_prev->getIntensity() < peak_height)
+      {
+        rt_apex = it->getMZ();
+      }
+      else if (it->getMZ() >= best_right && it_prev->getMZ() < best_right)
+      {
+        intensity_right = it->getIntensity();
+      }
+    }
+
+    double intensity_max, intensity_min, rt_min;
+    if (intensity_left >= intensity_right)
+    {
+      intensity_max = intensity_left;
+      intensity_min = intensity_right;
+      rt_min = best_right;
+    }
+    else 
+    {
+      intensity_max = intensity_right;
+      intensity_min = intensity_left;
+      rt_min = best_left;
+    }
+    // calculate the average noise level using the sin/cos rule
+    double delta_int = intensity_max - intensity_min;
+    double delta_rt = best_right - best_left;
+    double delta_rt_apex = std::fabs(rt_min-rt_apex);
+    double delta_int_apex = delta_int*delta_rt_apex/delta_rt;
+
+    avg_noise_level = intensity_min + delta_int_apex;
+    //NOTE: formula for calculating the background using the trapezoidal rule (future PR)
+    // background = intensity_min*delta_rt + 0.5*delta_int*delta_rt;
+
+    // calculate the background
+    background = 0.0;
+    for (MSChromatogram::const_iterator it = std::next(chromatogram.begin()); it != chromatogram.end(); ++it)
+    {
+      MSChromatogram::const_iterator it_prev = it;
+      --it_prev; //previous point
+
+      if (it->getMZ() >= best_left && it_prev->getMZ() < best_right)
+      {
+        // calculate the background using the formula
+        // y = mx + b where x = retention time, m = slope, b = left intensity
+        double delta_int = intensity_right - intensity_left; // sign will determine line direction
+        double delta_rt_current = (it->getMZ() - best_left);
+        double background_int_current = delta_int/delta_rt*delta_rt_current + intensity_left;
+        background = background + background_int_current;
```

Comment:
I guess this also works for non-uniformly spaced data while mine assumes the data is uniformly spaced? I think apart from that our code is using the same idea, right?

Reply:
The integration strategy for both methods is using the method of summing the intensities at each point.  Once the trapezoid integration is implemented as an alternative integration strategy, the code will need to be updated as we spoke about previously.

Reply:
Both methods have their pros and cons:

average: probably optimal for operating on the raw chromatogram as it will not be susceptible to spurious points right next to the peak boundaries.

exact: probably optimal for operating on the smoothed chromatogram as it will give a more precise calculating of the noise at the peak apex and the background under the peak.